task/orchestrator.go

```diff
@@ -200,6 +201,11 @@ func (ctx *OrchestrationContext) processEvent(e *backend.HistoryEvent) error {
 		// OrchestratorStarted is only used to update the current orchestration time
 		ctx.CurrentTimeUtc = e.Timestamp.AsTime()
 	} else if es := e.GetExecutionStarted(); es != nil {
+		// Extract source AppID from HistoryEvent Router if this is ExecutionStartedEvent
+		if e.GetRouter() != nil && e.GetRouter().GetSource() != "" {
```

Comment:
If router is not nil the surely source is always a non empty string?

If not, then `Source` should also be a pointer and we should check for non-nil for presence.

Please do not use empty string checks for presence- this should always be done on nil pointer checks.

```suggestion
		if e.GetRouter() != nil {
```